### ISSUE TITLE
Adjoint Solver: Optimization problem with an objective function dependent on separate simulations.

### ISSUE THREAD

**Issue by rafael-fuente** [NanoComp/meep#2198]
I'm currently working on an optimization project involving an objective function that is dependent on the DFT fields of completely independent simulations.

That is, denoting the DFT fields of a specific region of the simulation number $s$ by $\mathbf{E_s}$, I need the gradients with respect to the design region permittivity $\epsilon$ of an objective function of the form: 
$$\mathcal{F}(\mathbf{E_1} , \mathbf{E_2}, \mathbf{E_3} ... \mathbf{E_N} )$$

However, meep adjoint `OptimizationProblem` object only allows an objective function whose arguments are exclusive from a single simulation object.

Is there a way to handle my problem without substantially changing the meep adjoint module? 

One possibility, is computing the gradients in the following way:
$$\frac{\partial \mathcal{F}}{\partial \epsilon_i} =  \sum_{s} \frac{\partial \mathcal{F} }{\partial \mathbf{E_s}}   \frac{\partial \mathbf{E_s}}{\partial \epsilon_i} $$

Where the chain rule was used and $\frac{\partial \mathbf{E_s}}{\partial \epsilon_i}$ is the jacobian of the fields of the simulation number $s$.
How could I compute that Jacobian using meep adjoint module?

**Comment by ianwilliamson** [comment]
> I'm currently working on an optimization project involving an objective function that is dependent on the DFT fields of completely independent simulations.

This supported by the [Meep JAX wrapper](https://github.com/NanoComp/meep/blob/master/python/adjoint/wrapper.py). Essentially, you can create multiple `MeepJaxWrapper` objects wrapping separate simulations. You can then compose calls to these wrapper objects into any JAX loss function / computational graph that you want and then apply `jax.value_and_grad()`. The main limitation is that all of the involved simulations, although completely independent of each other, will be performed sequentially.
